packages/api/tests/app/test-middleware.js

```diff
@@ -0,0 +1,43 @@
+const test = require('ava');
+const express = require('express');
+const boom = require('express-boom');
+const request = require('supertest');
+
+const { EcsStartTaskError } = require('@cumulus/errors');
+
+const { asyncOperationEndpointErrorHandler } = require('../../app/middleware');
+
+const app = express();
+app.use(boom());
+
+// Set up fake endpoint to just test middleware
+app.post(
+  '/fake-async-endpoint',
+  () => {
+    throw new Error('failed to start');
+  },
+  asyncOperationEndpointErrorHandler
+);
+
+app.post(
+  '/fake-async-endpoint2',
+  () => {
+    throw new EcsStartTaskError('failed to start');
+  },
+  asyncOperationEndpointErrorHandler
+);
+
+test('request to replays endpoint returns 500 if starting ECS task throws unexpected error', async (t) => {
```

Comment:
Because the error handling for async operation endpoints is controlled by middleware, I also thought maybe we only need to test that middleware once (on fake endpoints), instead of writing separate tests each endpoint (/replays, bulk granules, etc) to see whether they handle the errors correctly. But is that a good assumption? 

All this proves is that the middleware will return the correct response if it's the only middleware in the chain, but the endpoint routes could be defined with other middleware